ext/pdo_firebird/firebird_driver.c

```diff
@@ -1216,6 +1216,18 @@ static int pdo_firebird_get_attribute(pdo_dbh_t *dbh, zend_long attr, zval *val)
 }
 /* }}} */
 
+#if FB_API_VER >= 30
+/* called by PDO to check liveness */
+static zend_result pdo_firebird_check_liveness(pdo_dbh_t *dbh) /* {{{ */
+{
+	pdo_firebird_db_handle *H = (pdo_firebird_db_handle *)dbh->driver_data;
+
+	/* fb_ping return 0 if the connection is alive */
+	return fb_ping(H->isc_status, &H->db) ? FAILURE : SUCCESS;
+}
+/* }}} */
+#endif
```

Comment:
I downloaded the source from the following list and checked the included `ibase.h`:
https://firebirdsql.org/en/discontinued-versions/

It turns out that this feature was introduced since v3.0.